src/traces/scatter/fill_hover_text.js

```diff
@@ -0,0 +1,36 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../../lib');
+
+/** Fill hover 'pointData' container with 'correct' hover text value
+ *
+ * - If trace hoverinfo contains a 'text' flag and hovertext is not set,
+ *   the text elements will be seen in the hover labels.
+ *
+ * - If trace hoverinfo contains a 'text' flag and hovertext is set,
+ *   hovertext takes precedence over text
+ *   i.e. the hoverinfo elements will be seen in the hover labels
+ *
+ *  @param {object} calcPt
+ *  @param {object} trace
+ *  @param {object || array} contOut (mutated here)
+ */
+module.exports = function fillHoverText(calcPt, trace, contOut) {
+    var fill = Array.isArray(contOut) ?
+        function(v) { contOut.push(v); } :
+        function(v) { contOut.text = v; };
+
+    var htx = Lib.extractOption(calcPt, trace, 'htx', 'hovertext');
+    if(htx && htx !== '') return fill(htx);
+
+    var tx = Lib.extractOption(calcPt, trace, 'tx', 'text');
+    if(tx && tx !== '') return fill(tx);
+};
```

Comment:
Nice! 🌴  Does this really belong in `scatter` though? Perhaps in `fx`?

Reply:
I have no strong opinion here. This routine felt similar to 

https://github.com/plotly/plotly.js/blob/master/src/traces/scatter/arrays_to_calcdata.js

that's why I put it in `traces/scatter/`

Reply:
That's fine, we certainly have a long history of traces pulling from each other, particularly from scatter, and there *is* something nice about it being in the `traces` dir. Perhaps at some point we introduce a `traces/common` dir for stuff like this but leave it as is for now.